test/testvideocapture.c

```diff
@@ -0,0 +1,787 @@
+
+    renderer = SDL_CreateRenderer(window, NULL, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);
+    if (renderer == NULL) {
+        SDL_Log("Couldn't create renderer: %s", SDL_GetError());
+        return 1;
+    }
+
+#if USE_DMABUF
+
+#define LOAL_FUNC(f)                            \
+    f = (f ##_t)SDL_GL_GetProcAddress(#f);      \
+    if (f == NULL) {                            \
+        SDL_Log("Cannot load function: " #f );  \
+        return 0;                               \
+    }                                           \
+
+    LOAL_FUNC(eglGetError)
+    LOAL_FUNC(eglGetCurrentDisplay)
+    LOAL_FUNC(eglCreateImageKHR)
+    LOAL_FUNC(eglDestroyImageKHR)
+    LOAL_FUNC(glActiveTexture)
+    LOAL_FUNC(glEGLImageTargetTexture2DOES)
+
+    display = eglGetCurrentDisplay();
+    if (display == EGL_NO_DISPLAY) {
+        SDL_Log("no dispay");
+        return 0;
+    }
+#endif
+
+
+    SDL_LogSetAllPriority(SDL_LOG_PRIORITY_VERBOSE);
+
+    id = SDL_OpenVideoCapture(NULL);
+
+    if (id == SDL_VIDEO_CAPTURE_INVALID) {
+        SDL_Log("Error SDL_OpenVideoCapture: %s", SDL_GetError());
+    }
+
+    {
+        /* List formats */
+        int i, num = SDL_GetNumVideoCaptureFormats(id);
+        for (i = 0; i < num; i++) {
+            Uint32 format, type;
+            SDL_GetVideoCaptureFormat(id, i, &format, &type);
+            SDL_Log("format %d/%d type=%08" SDL_PRIx32 ": %s", i, num, type, SDL_GetPixelFormatName(format));
+            {
+                Uint32 w, h;
+                int j, num2 = SDL_GetNumVideoCaptureFrameSizes(id, format);
+                for (j = 0; j < num2; j++) {
+                    SDL_GetVideoCaptureFrameSize(id, format, j, &w, &h);
+                    SDL_Log("  framesizes %d/%d :  %" SDL_PRIu32 " x %" SDL_PRIu32 "", j, num2, w, h);
+                }
+            }
+        }
+    }
+
+    /* Set Spec */
+    {
+        int ret;
+        /* forced_format */
+        SDL_VideoCaptureSpec desired;
+        SDL_zero(desired);
+        desired.width = 640 * 2;
+        desired.height = 360 * 2;
+        ret = SDL_SetVideoCaptureSpec(id, &desired, &obtained, SDL_VIDEO_CAPTURE_ALLOW_ANY_CHANGE);
+
+        if (ret < 0) {
+            SDL_SetVideoCaptureSpec(id, NULL, &obtained, 0);
+        }
+    }
+
+    SDL_Log("Open capture video device. Obtained spec: size=%" SDL_PRIu32 "x%" SDL_PRIu32 " format=%s",
+            obtained.width, obtained.height, SDL_GetPixelFormatName(obtained.format));
+
+    {
+        SDL_VideoCaptureSpec spec;
+        if (SDL_GetVideoCaptureSpec(id, &spec) == 0) {
+            SDL_Log("Read spec: size=%" SDL_PRIu32 "x%" SDL_PRIu32 " format=%s",
+                    spec.width, spec.height, SDL_GetPixelFormatName(spec.format));
+        } else {
+            SDL_Log("Error read spec: %s", SDL_GetError());
+        }
+    }
+
+    if (SDL_StartVideoCapture(id) < 0) {
+        SDL_Log("error SDL_StartVideoCapture(): %s", SDL_GetError());
```

Comment:
Same question here. Should it quit?

Reply:
yes, maybe. app could wait that you plug a usb camera